spec/models/upload_spec.rb

```diff
@@ -47,6 +47,156 @@
+        allow(FileHelper).to receive(:is_supported_video?).with("small.mp4").and_return(false)
+
+        upload = Fabricate(:upload, original_filename: "small.mp4", extension: "mp4", user: user)
+
+        expect(Jobs).not_to have_received(:enqueue).with(:convert_video, upload_id: upload.id)
+      end
+
+      it "does not enqueue a convert_video job when video conversion is disabled" do
+        SiteSetting.video_conversion_enabled = false
+        allow(FileHelper).to receive(:is_supported_video?).with("small.mp4").and_return(true)
+
+        upload = Fabricate(:upload, original_filename: "small.mp4", extension: "mp4", user: user)
+
+        expect(Jobs).not_to have_received(:enqueue).with(:convert_video, upload_id: upload.id)
+      end
+
+      it "does not enqueue a convert_video job when S3 uploads are disabled" do
+        SiteSetting.enable_s3_uploads = false
+        allow(FileHelper).to receive(:is_supported_video?).with("small.mp4").and_return(true)
+
+        upload = Fabricate(:upload, original_filename: "small.mp4", extension: "mp4", user: user)
+
+        expect(Jobs).not_to have_received(:enqueue).with(:convert_video, upload_id: upload.id)
+      end
+
+      it "does not enqueue a convert_video job for non-video files" do
+        upload = Fabricate(:upload, original_filename: "image.png", extension: "png", user: user)
+
+        expect(Jobs).not_to have_received(:enqueue).with(:convert_video, upload_id: upload.id)
+      end
+
+      it "does not enqueue a convert_video job if OptimizedVideo already exists" do
+        allow(FileHelper).to receive(:is_supported_video?).with("small.mp4").and_return(true)
+        allow(FileHelper).to receive(:is_supported_video?).with("video_converted.mp4").and_return(
+          true,
+        )
+
+        # Create original upload
+        upload = Fabricate(:upload, original_filename: "small.mp4", extension: "mp4", user: user)
+
+        # Create OptimizedVideo record for the original upload
+        optimized_video = Fabricate(:optimized_video, upload: upload)
+
+        # Reset the job spy to clear previous calls and re-stub
+        RSpec::Mocks.space.proxy_for(Jobs).reset
+        allow(Jobs).to receive(:enqueue)
+
+        # Update the original upload to trigger after_commit
+        upload.update!(filesize: 12_345)
+
+        # The original upload should not enqueue a job because OptimizedVideo already exists
+        expect(Jobs).not_to have_received(:enqueue).with(:convert_video, upload_id: upload.id)
+      end
+
+      it "does not enqueue a convert_video job on update, only on create" do
+        allow(FileHelper).to receive(:is_supported_video?).with("small.mp4").and_return(true)
+
+        upload = Fabricate(:upload, original_filename: "small.mp4", extension: "mp4", user: user)
+
+        # Reset the job spy to clear previous calls and re-stub
+        RSpec::Mocks.space.proxy_for(Jobs).reset
+        allow(Jobs).to receive(:enqueue)
```

Comment:
Related to https://github.com/discourse/discourse/pull/35013/files#r2390084976, I believe we don't need these lines if we were to use the SidekiqHelpers.